test/e2e/kustomizations.go

```diff
@@ -0,0 +1,168 @@
+	workDir, err := os.MkdirTemp("", "kictest.")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(workDir)
+	orig, err := io.ReadAll(baseManifestReader)
+	if err != nil {
+		return nil, err
+	}
+	err = os.WriteFile(filepath.Join(workDir, "base.yaml"), orig, 0o600)
+	if err != nil {
+		return nil, err
+	}
+	kustomization.Bases = []string{"base.yaml"}
+	marshalled, err := yaml.Marshal(kustomization)
+	if err != nil {
+		return nil, err
+	}
+	err = os.WriteFile(filepath.Join(workDir, "kustomization.yaml"), marshalled, 0o600)
+	if err != nil {
+		return nil, err
+	}
+	kustomized, err := runKustomize(workDir)
+	if err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(kustomized), nil
+}
+
+// patchControllerImage replaces the kong/kubernetes-ingress-controller image with the provided image and tag,
+// and returns the modified manifest.
+func patchControllerImage(baseManifestReader io.Reader, image, tag string) (io.Reader, error) {
+	kustomization := types.Kustomization{
+		Bases: []string{"base.yaml"},
+		Images: []types.Image{
+			{
+				Name:    "kong/kubernetes-ingress-controller",
+				NewName: image,
+				NewTag:  tag,
+			},
+		},
+	}
+	return getKustomizedManifest(baseManifestReader, kustomization)
+}
+
+// patchKongImage replaces the kong and kong/kong-gateway images in a manifest with the provide image and tag,
+// and returns the modified manifest.
+func patchKongImage(baseManifestsReader io.Reader, image, tag string) (io.Reader, error) {
+	kustomization := types.Kustomization{
+		Bases: []string{"base.yaml"},
+		Images: []types.Image{
+			{
+				Name:    "kong/kong-gateway",
+				NewName: image,
+				NewTag:  tag,
+			},
+			{
+				Name:    "kong",
+				NewName: image,
+				NewTag:  tag,
+			},
+		},
+	}
+	return getKustomizedManifest(baseManifestsReader, kustomization)
+}
+
+// patchControllerStartTimeout adds or updates the controller container CONTROLLER_KONG_ADMIN_INIT_RETRIES and
+// CONTROLLER_KONG_ADMIN_INIT_RETRY_DELAY environment variables with the provided values, and returns the modified
+// manifest.
+func patchControllerStartTimeout(baseManifestReader io.Reader, tries int, delay string) (io.Reader, error) {
+	kustomization := types.Kustomization{
+		Bases: []string{"base.yaml"},
+		Patches: []types.Patch{
+			{
+				Patch: fmt.Sprintf(initRetryPatch, delay, tries),
+				Target: &types.Selector{
+					ResId: resid.ResId{
+						Gvk: resid.Gvk{
+							Group:   "apps",
+							Version: "v1",
+							Kind:    "Deployment",
+						},
+						Name:      "ingress-kong",
+						Namespace: "kong",
+					},
+				},
+			},
+		},
+	}
+	return getKustomizedManifest(baseManifestReader, kustomization)
+}
+
+// patchLivenessProbes patches the given container's liveness probe, replacing the initial delay, period, and failure
+// threshold.
+func patchLivenessProbes(baseManifestReader io.Reader, container, initial, period, failure int) (io.Reader, error) {
```

Comment:
I know that this might look a lot easier on the eyes like this but I'd much rather see us use `time.Duration` for expressing time duration (which can then extract seconds via: `.Seconds()`).

Especially given the fact that there's a `container` argument which has the same type just next to the duration arguments 😬